nacl-fast.js

```diff
@@ -1,3 +1,5 @@
+var nacl = {};
```

Comment:
This unfortunate pollutes global scope. Perhaps, I should move all this modularization stuff into a a build step, like I did here: https://github.com/dchest/fast-sha256-js

Reply:
I don't quite understand what you did in fast-sha-256-js though. I thought exports were reserved for the top level as they are statically resolved, not dynamic. Would that still work as a direct import in a browser?

Reply:
Yeah, they are for CommonJS, but this can also be used when directly included using <script> tag.

Reply:
I meant for ESM (edited my comment before I saw you reply), can ESM `export` be dynamic or non-top-level?

Reply:
Ah, no I don't think it can be dynamic.

Reply:
My hope is to be able to use this in a browser without any build step, so if that is possible while still keeping test and code duplication at an okay amount that'd be awesome.

If not I'll make it so some other way :slightly_smiling_face: 
Thanks for looking at and considering this!

Reply:
Indeed, thanks! I think I'll make a special build step that will insert those things and generate .mjs files.